relay-server/src/actors/outcome.rs

```diff
@@ -166,145 +169,147 @@ pub enum DiscardReason {
+        }
+    }
+
+    fn to_outcome_id(&self) -> u8 {
+        match self {
+            Outcome::Accepted => 0,
+            Outcome::Filtered(_) => 1,
+            Outcome::RateLimited(_) => 2,
+            Outcome::Invalid(_) => 3,
+            Outcome::Abuse => 4,
+        }
+    }
+
+    fn to_reason(&self) -> Option<&str> {
+        match self {
+            Outcome::Accepted => None,
+            Outcome::Invalid(discard_reason) => Some(discard_reason.name()),
+            Outcome::Filtered(filter_key) => Some(filter_key.name()),
+            Outcome::RateLimited(code_opt) => code_opt.as_ref().map(|code| code.as_str()),
+            Outcome::Abuse => None,
+        }
+    }
+}
+
+impl DiscardReason {
+    pub fn name(self) -> &'static str {
+        match self {
+            DiscardReason::Duplicate => "duplicate",
+            DiscardReason::ProjectId => "project_id",
+            DiscardReason::AuthVersion => "auth_version",
+            DiscardReason::AuthClient => "auth_client",
+            DiscardReason::NoData => "no_data",
+            DiscardReason::TooLarge => "too_large",
+            DiscardReason::DisallowedMethod => "disallowed_method",
+            DiscardReason::ContentType => "content_type",
+            DiscardReason::MultiProjectId => "multi_project_id",
+            DiscardReason::MissingMinidumpUpload => "missing_minidump_upload",
+            DiscardReason::InvalidMinidump => "invalid_minidump",
+            DiscardReason::SecurityReportType => "security_report_type",
+            DiscardReason::SecurityReport => "security_report",
+            DiscardReason::Cors => "cors",
+            DiscardReason::ProcessUnreal => "process_unreal",
+
+            // Relay specific reasons (not present in Sentry)
+            DiscardReason::Payload => "payload",
+            DiscardReason::InvalidJson => "invalid_json",
+            DiscardReason::InvalidMultipart => "invalid_multipart",
+            DiscardReason::InvalidMsgpack => "invalid_msgpack",
+            DiscardReason::InvalidTransaction => "invalid_transaction",
+            DiscardReason::InvalidEnvelope => "invalid_envelope",
+            DiscardReason::ProjectState => "project_state",
+            DiscardReason::DuplicateItem => "duplicate_item",
+            DiscardReason::Internal => "internal",
+        }
+    }
+}
+
+/// The outcome message is serialized as json and placed on the Kafka topic using OutcomePayload
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct OutcomePayload {
+    /// The timespan of the event outcome.
+    timestamp: String,
+    /// Organization id.
+    org_id: Option<u64>,
+    /// Project id.
+    project_id: ProjectId,
+    /// The DSN project key id.
+    key_id: Option<u64>,
+    /// The outcome.
+    outcome: u8,
+    /// Reason for the outcome.
+    reason: Option<String>,
+    /// The event id.
+    event_id: Option<EventId>,
+    /// The client ip address.
+    remote_addr: Option<String>,
```

Comment:
Since this now goes over the wire, please mark all of the `Option<>` attributes with `#[serde(default, skip_serializing_if="Option::is_none")]` to reduce the payload size. We can double-check that this doesn't cause any troubles in the consumer.